### PR TITLE
chore(i18n): improve performance of language file loading

### PR DESCRIPTION
**Pull request by hypeJunction**
Eliminates redudant loops through directories when languages are set.
Cache loaded files to avoid multiple inclusions during testing

### CODE REVIEW

File: engine/classes/Elgg/I18n/Translator.php [hunk 1]
```diff
@@ -402,33 +403,52 @@ public function registerTranslations($path, $load_all = false, $language = null)
...
 
+	/**
+	 * Load cached or include a language file by its path
+	 *
+	 * @param string $path Path to file
+	 * @return bool
+	 */
+	protected function includeLanguageFile($path) {
+		$cache_key = "lang/" . sha1($path);
+		$result = elgg_get_system_cache()->load($cache_key);
```
Comment (mrclay) [inline engine/classes/Elgg/I18n/Translator.php]: Isn't this more costly than the file read?

Comment (hypeJunction) [reply engine/classes/Elgg/I18n/Translator.php]: Not if there is persistent cache. When it's a file system cache, I think it's the same thing.

Comment (hypeJunction) [reply engine/classes/Elgg/I18n/Translator.php]: On an average page load this won't have much effect, on tests however it gives a decent boost.

Comment (jdalsem) [reply engine/classes/Elgg/I18n/Translator.php]: I see no effect on test runtime...

Comment (hypeJunction) [reply engine/classes/Elgg/I18n/Translator.php]: Try it locally

Comment (hypeJunction) [reply engine/classes/Elgg/I18n/Translator.php]: And travis is no indicator, build times have been incredibly volatile

Comment (hypeJunction) [reply engine/classes/Elgg/I18n/Translator.php]: Also this is what fixed the phpdbg memory issues.

**Comment by mrclay** [comment]
On my VM my tests run out of memory at 2.2GB at 98%. I haven't been able to finish tests in awhile.

**Comment by hypeJunction** [comment]
If I run the whole suite that's about 2-3 Gb in cca 10 minutes, hence I am trying to speed things up. I have wasted so much time waiting for test results.

**Comment by jdalsem** [comment]
this is me hoping for quicker tests... merging now

**Pull request merged**